### ISSUE TITLE
RAML nillable shorthand types not recognized in uriParameters/queryParameters/headers maps

### ISSUE THREAD

**Issue by kdubb** [aml-org/amf#849]
Your issue may already be reported! Please search on Github issues before creating one.

* **I'm submitting a ...**
  - [x] bug report
  - [ ] feature request
  - [ ] question
  - [ ] other

* **What is the current behavior?**

Using a nillable shorthand type (aka `string?`) inside the headers parameter map produces the error `Cannot declare unresolved parameter`.

* **If the current behavior is a bug, please provide the minimal steps to reproduce with AMF code. Please don't upload AMF code mixed with yours.**

The following RAML snippet(s) produce the described error:

#### URI Parameters
```
/tests/{str1}/{str2}:
  get:
    displayName: fetchTest
    uriParameters:
      str1: string?         ### Error on this line
      str2: string | nil
    responses:
      200:
        body:
          type: string
```

#### Query Parameters
```
/tests:
  get:
    displayName: fetchTest
    queryParameters:
      str1: string?         ### Error on this line
      str2: string | nil
    responses:
      200:
        body:
          type: string
```

#### Headers
```
/tests:
  get:
    displayName: fetchTest
    headers:
      str1: string?         ### Error on this line
      str2: string | nil
    responses:
      200:
        body:
          type: string
```

* **What is the expected behavior?**

Nillable shorthand types should work as they do in other places.

* **What is the motivation/use case for changing the behavior? (for feature requests)**

* **Please tell us about your environment:**
  
  - AMF Version: 4.7.0
  - AMF Distribution: JVM
  - Operating System: macOS
  - Link to your project: N/A

* **Other information** (e.g. detailed explanation, stack traces, related issues, suggestions how to fix, links for us to have context, eg. StackOverflow, Gitter, etc)

**Comment by looseale** [comment]
Hi @kdubb. I created a bug in our internal issue system to track this: https://www.mulesoft.org/jira/browse/APIMF-2934. I wrote this here to link our internal ticket with Github.

All updates on this issue will be communicated through this channel (the Github issue).
Thanks!

**Comment by tomsfernandez** [comment]
Hi @kdubb. This issue was fixed in PR #875 and will be available in AMF's next release scheduled in 2 weeks. 

Thanks for raising the issue.
